src/request.rs

```diff
@@ -407,6 +419,16 @@ impl<B> RequestBuilder<B> {
         self.accept_invalid_hostnames = accept_invalid_hostnames;
         self
     }
+
+    /// Sets the TLS client configuration
+    ///
+    /// Defaults to a configuration using the root certificates
+    /// from the webpki-roots crate.
+    #[cfg(feature = "tls-rustls")]
+    pub fn client_config(mut self, client_config: Arc<ClientConfig>) -> Self {
```

Comment:
Would it be a good idea to take a `impl Into<Arc<ClientConfig>>` to make the boxing to Arc automatic if you don't intend to share the structure?